synapse/api/errors.py

```diff
@@ -505,17 +510,22 @@ class LimitExceededError(SynapseError):
 
     def __init__(
         self,
+        limiter_name: str,
         code: int = 429,
-        msg: str = "Too Many Requests",
         retry_after_ms: Optional[int] = None,
         errcode: str = Codes.LIMIT_EXCEEDED,
     ):
-        super().__init__(code, msg, errcode)
+        super().__init__(code, "Too Many Requests", errcode)
         self.retry_after_ms = retry_after_ms
+        self.limiter_name = limiter_name
 
     def error_dict(self, config: Optional["HomeServerConfig"]) -> "JsonDict":
         return cs_error(self.msg, self.errcode, retry_after_ms=self.retry_after_ms)
 
+    @property
+    def debug_context(self) -> Optional[str]:
+        return self.limiter_name
```

Comment:
`debug_context` could probably just be a class-property too and `__init__` could set `self.debug_context = limiter_name`. I don't feel strongly either way though.

Reply:
No strong feelings here either. Suggest we leave it as-is in the interests of landing this sooner rather than later.